src/login/pam_systemd.c

```diff
@@ -1491,6 +1489,46 @@ static int export_legacy_dbus_address(
         return update_environment(handle, "DBUS_SESSION_BUS_ADDRESS", t);
 }
 
+static int setup_runtime_directory(
+                pam_handle_t *handle,
+                UserRecord *ur,
+                const char *runtime_directory,
+                const char *area) {
+
+        int r;
+
+        assert(handle);
+        assert(ur);
+
+        if (!runtime_directory) {
+                /* If this is an area switch request, always reset $XDG_RUNTIME_DIR if we got nothing
+                 * to ensure the main runtime dir won't be clobbered. */
+                if (area)
+                        return update_environment(handle, "XDG_RUNTIME_DIR", NULL);
+
+                return PAM_SUCCESS;
+        }
+
+        /* Also create a per-area subdirectory for $XDG_RUNTIME_DIR, so that each area has their own
+         * set of runtime services. We follow the same directory structure as for $HOME. Note that we
+         * do not define any form of automatic clean-up for the per-aera subdirs beyond the regular
```

Comment:
"area", but this is a preexisting condition, so not a big deal.

Reply:
-> 0140f260462c2ff1a783609322e39de0a17b6574